Better DNS records

Bouska opened the pull request:
The script now outputs a DNS record with all of its public IP + a DMARC record line.

Bouska committed:
[fix] Get a single IP adress 

tun0 can have multiples IP address, in particular IPv6 address (the one leased by the VPN provider and one from your IP subnet). Currently the script returns a list of all the IP it found. It should only return the first (?) one.

Bouska committed:
[ehn] Add DMARC record

Add a DMARC record. From Wikipedia, "a DMARC policy allows a sender's domain to indicate that their emails are protected by SPF and/or DKIM, and tells a receiver what to do if neither of those authentication methods passes". Proposed policy is to log affected messages and send daily reports to postmaster@DOMAIN.TLD.

Bouska committed:
[fix] Handle multiple IP or no IP

Create as much record as there are IPs. Handle multiple IP (e.g. with IPv6 subnet + lease) or even no IP (e.g. full IPv6 Cube).
Revert 4f43e57.

Bouska committed:
[fix] SPF record in a single line

I forgot that echo command outputs a trailing newline. Using -n option to avoid that

Psycojoker commented:
I might be missing something but in which situation could we have several values in ip4 and ipv6?

Bouska commented:
Hu, good question. It might be a glitch on my side, sometimes "ip addr show tun0" gives me two IPv6: one of my IP subnet and the one leased by Neutrinet. If that is just a bug, just discard my PR

jvaubourg commented:
This is not a bug, you can have both an "end-point" address and an address derivated from your IPv6 delegated prefix (ending by "::42") at the same time, on tun0. But, when the second one is available, you should only create a AAAA for it.

Moreover, once Wifi Hotspot is installed, the address derivated from your IPv6 delegated prefix moves to "wlan0" (or another interface corresponding to the wifi antenna). But, in this case too, you should create a AAAA only for this address.

To sum up, if you find an address ending by "::42" on any interface, you should use it. Else, you should use the IPv6 address found on tun0.

Psycojoker commented:
@jvaubourg can I let you handle this PR then (choosing to merge it or not etc...)? You are better suited than me for that.

jvaubourg commented:
@Bouska As explained in my previous comment, we do not have to handle any case where there are several IPv6 or IPv4. So, rather than adding a loop, we should fix the $ip6 and $ip4 generation.

According to the script, I understand that providing a value for $ip6_net is mandatory. So, finding the IPv6 to use is very easy: `ip6="${ip6_net}42"`. And for IPv4, we should just restrict the awk output to its first line.

Can I let you fix your PR, in this way? Thank you a lot :).

Bouska committed:
[enh] Add DKIM option
Bring back the option to install/configure DKIM during installation. Inspired from https://yunohost.org/#/dkim. Not tested (yet)
